src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -19,6 +20,11 @@ public IBuilder Builder
             }
         }
 
+        public Task RaiseNotification<T>(T notification) where T : INotification
+        {
+            return Builder.Build<EventAggregator>().Raise(notification);
```

Comment:
I think we can resolve it once on the `RootContext` and then fetch it like we do for the `Builder` property 10 lines above (I think this is the same as you mentioned with using the extension dictionary?).